examples/Makefile

```diff
@@ -1,81 +1,68 @@
 
-default: all
-
 # This Makefile demonstrates different options for using the AbcSmc library
-# with different approaches to the simulator, as well as a toy simulator for use with AbcSmc.
+# with different approaches to the simulator, illustrated with a toy simulator.
 #
 # The toy simulator represents rolling a number dice, all of the same face count, then returns their
 # sum and standard deviation. The simulator is defined in dice.h, and follows the function signature
 # required by AbcSmc.
 #
-# AbcSmc supports three different ways to use a simulator:
+# AbcSmc supports several ways to use a simulator:
 #
 # 1. The simulator is compiled into an executable along with the library (main_sql.cpp --> abc_sql).
 # 2. The simulator is compiled into a shared library (dice.cpp / dice.so) and loaded dynamically (main_dynamic.cpp --> abc_dynamic).
 # 3. The simulator is compiled into a separate executable (dice_game.cpp) and invoked externally (main_exec.cpp --> abc_exec).
 #
 # This Makefile demonstrates how to compile and run each of these options.
 
-# make variable definitions
 CPP := g++
 
 CFLAGS := -O2 -Wall -std=c++17 --pedantic
 SHAREDFLAGS := $(CFLAGS) -fPIC -shared -rdynamic -Wl,--no-as-needed
+LIBS := -lm -lgsl -lgslcblas -ldl
 
-MKFILE_PATH = $(shell dirname $(realpath $(lastword $(MAKEFILE_LIST))))
-ABCDIR := $(MKFILE_PATH)/..
-SQLDIR := $(ABCDIR)/sqdb
-JSNDIR := $(ABCDIR)/jsoncpp/include
-EIGDIR := $(ABCDIR)/pls/eigen
-GSL_LIB ?= -lm -lgsl -lgslcblas -pthread
+DEMOS := direct executable shared integral
 
-ABC_INC := -I$(ABCDIR) -I$(SQLDIR) -I$(JSNDIR) -I$(EIGDIR) -L$(ABCDIR) -labc $(GSL_LIB) -ldl
+default: $(patsubst %,demo/%,$(DEMOS)) $(patsubst %,%/config.json,$(DEMOS)) dice.so
 
-#ifdef TACC_GSL_INC
-#GSL_INC = -I$$TACC_GSL_INC
-#endif
-#ifdef HPC_GSL_INC
-#GSL_INC = -I$$HPC_GSL_INC
-#endif
+bin output:
+	mkdir -p $@
 
-DEMOS := sql dynamic exec
+demo:
+	cmake -S . -B $@
 
-all: ../libabc.a dice_game dice.so $(addprefix abc_,$(DEMOS))
+demo/%: demo
+	cd $^ && $(MAKE) $*
 
-abc_%.json: reference.json %_partial.json
+%/config.json: reference.json %/partial.json
 	gojq -s add $^ > $@
```

Comment:
This shouldn't be a dependency for "hello world" examples . . . or cmake needs to install it, or something.  IDK, I'm not sure we want it as a dependency for testing, either, if we expect other people to be able to run tests.  Is gojq available across platforms?